src/cpr-tooltip.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+
+export default class CprTooltip extends React.Component {
+	componentDidMount() {
+		if (this.refs.child) {
+			ensureEventListeners.call(this);
+		}
+	}
+
+	componentWillReceiveProps() {
+		if (this.refs.child) {
+			ensureEventListeners.call(this)
+		}
+	}
+
+	render() {
+		let {children} = this.props;
+
+		children = React.cloneElement(children, {
+			ref: `child`
+		});
+
+		if (this.tooltipEl && this.refs.child) {
+			updateTooltip.call(this);
+		}
+
+		return children;
+	}
+
+	componentWillUnmount() {
+		cleanupTooltip.call(this);
+	}
+}
+
+function cleanupTooltip() {
+	if (this.timeout) {
+		clearTimeout(this.timeout);
+		delete this.timeout;
+	}
+
+	if (this.tooltipEl) {
+		if (this.tooltipEl.parentNode) {
+			this.tooltipEl.parentNode.removeChild(this.tooltipEl);
+		}
+
+		delete this.tooltipEl;
+	}
+}
+
+function ensureEventListeners() {
+	const el = this.refs.child;
+
+	if (!el.classList.contains('cpr-tooltip-target')) {
+		el.addEventListener('mouseover', mousedOver.bind(this))
+		el.addEventListener('mouseleave', cleanupTooltip.bind(this))
+
+		el.classList.add('cpr-tooltip-target');
+	}
+}
+
+function mousedOver() {
+	if (!this.timeout) {
+		this.timeout = setTimeout(() => {
+			this.timeout = null;
+
+			if (!this.tooltipEl) {
+				this.tooltipEl = document.createElement('span');
+				this.tooltipEl.classList.add('cpr-tooltip-text')
+				document.body.appendChild(this.tooltipEl);
+			}
+
+			updateTooltip.call(this, this.tooltipEl, this.refs.child);
```

Comment:
Oh good point -- I used to use them, but realized that inside of the function I can just get them off of the `this`. I'll remove it